src/classes/RD2_VisualizeScheduleController.cls

```diff
@@ -61,74 +61,224 @@ public with sharing class RD2_VisualizeScheduleController {
             for (RD2_ScheduleService.ActiveSchedule schedule : scheduleService.getVisualizedSchedules(rd)) {
-                activeSchedules.add(new Schedule(counter++, rd, schedule));
+                dataTable.addRecord(new Schedule(counter++, schedule));
             }
         }
 
-        if (activeSchedules.isEmpty()) {
+        if (dataTable.getRecords().isEmpty()) {
             throw new AuraHandledException(System.Label.RD2_ScheduleVisualizerMessageNoActiveSchedule);
         }
 
-        return activeSchedules;
+        sanitizeSchedules(dataTable);
+
+        return new Map<String, Object>{ 'dataTable' => dataTable };
+    }
+
+
+    /**
+     * @description Constructs the data table columns for the projected installments display
+     * @param rd Recurring Donation
+     * @return DataTable
+     */
+    @TestVisible
+    private static DataTable buildInstallmentDataTable(npe03__Recurring_Donation__c rd) {
+        return new DataTable()
+            .addColumn(new DataTableColumn()
+                .withFieldName('donationDate')
+                .withLabel(System.Label.RD2_ScheduleVisualizerColumnDate)
+                .withIsAccessible(Schema.SObjectType.npe03__Recurring_Donation__c.fields.Day_Of_Month__c)
+                .withType(Schema.DisplayType.DATE)
+                .withTypeAttributes(new Map<String, Object>{
+                    'month' => '2-digit',
+                    'day' => '2-digit'
+                })
+            )
+            .addColumn(new DataTableColumn(Schema.SObjectType.npe03__Recurring_Donation__c.fields.npe03__Amount__c)
+                .withFieldName('amount')
+                .withTypeAttributes(new Map<String, Object>{
+                    'currencyCode' => getCurrencyCode(rd)
+                })
+            )
+            .addColumn(new DataTableColumn(Schema.SObjectType.npe03__Recurring_Donation__c.fields.PaymentMethod__c)
+                .withFieldName('paymentMethod')
+            );
+    }
+
+    /**
+     * @description Constructs the data table RD schedule columns 
+     * @param rd Recurring Donation
+     * @return DataTable
+     */
+    @TestVisible
+    private static DataTable buildScheduleDataTable(npe03__Recurring_Donation__c rd) {
+        return new DataTable()
+            .addColumn(new DataTableColumn()
+                .withFieldName('isCurrent')
+                .withType(Schema.DisplayType.BOOLEAN)
+                .withIsAccessible(true)
+            )
+            .addColumn(new DataTableColumn(Schema.SObjectType.npe03__Recurring_Donation__c.fields.npe03__Amount__c)
+                .withFieldName('amount')
+                .withTypeAttributes(new Map<String, Object>{
+                    'currencyCode' => getCurrencyCode(rd)
+                })
+            )
+            .addColumn(new DataTableColumn(Schema.SObjectType.npe03__Recurring_Donation__c.fields.PaymentMethod__c)
+                .withFieldName('paymentMethod')
```

Comment:
The field name in the data table is the SObject field name or the Object field name. In our case, the field name is the Installment or the Schedule Object field name, the value needs to be provided additionally. I can put values into constants if that will make code readable and less error prone?

Reply:
Yeah. Basically that's my concern. For example, is "paymentmethod" equivalent to "PaymentMethod"; and what if someone types "paymetnmethod" instead. Seems like it would be very easy to make a mistake and break the page.